From ff3f153b514601785935df48f4cdabf0e5fef091 Mon Sep 17 00:00:00 2001
From: Luca Di Maio <luca.dimaio1@gmail.com>
Date: Wed, 12 Nov 2025 09:53:03 +0100
Subject: [PATCH v5] fix: label /var/log/faillock accordingly

as per STIG: xccdf_org.ssgproject.content_rule_account_password_selinux_faillock_dir
the /var/log/faillock dir must be labeled faillog_t

by default we only have /run/faillock (and /var/run/faillock) labeled,
but STIG: xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_dir
requires the faillock dir to be in /var/log.

Fixes VMS-162

Signed-off-by: Luca Di Maio <luca.dimaio1@gmail.com>
---
 policy/modules/system/authlogin.fc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/authlogin.fc b/policy/modules/system/authlogin.fc
index 2dd492703..375777ab2 100644
--- a/policy/modules/system/authlogin.fc
+++ b/policy/modules/system/authlogin.fc
@@ -60,6 +60,7 @@ ifdef(`distro_suse', `
 /var/log/btmp.*		--	gen_context(system_u:object_r:faillog_t,s0)
 /var/log/dmesg		--	gen_context(system_u:object_r:var_log_t,s0)
 /var/log/faillog.*	--	gen_context(system_u:object_r:faillog_t,s0)
+/var/log/faillock.*	--	gen_context(system_u:object_r:faillog_t,s0)
 /var/log/lastlog.*	--	gen_context(system_u:object_r:lastlog_t,s0)
 /var/log/syslog		--	gen_context(system_u:object_r:var_log_t,s0)
 /var/log/tallylog.*	--	gen_context(system_u:object_r:faillog_t,s0)
-- 
2.51.2

